src/analysis.ts

```diff
@@ -110,3 +111,78 @@ export function intervalMatrix(scale: Scale) {
   }
   return result
 }
+
+// Interval matrix a.k.a the modes of a scale, but with step vectors.
+// Return the step vector for the j-step on mode i.
+export function stepVectorMatrix(scale: Scale): [[[[String, number]]]] {
+  console.log('stepVectorMatrix was called')
+  const result = []
+  const columns = [...Array(scale.size + 1).keys()]
+  for (let i = 0; i < scale.size; ++i) {
+    const mode = scale.rotate(i)
+    result.push(columns.map((j) => getStepVector(mode, j)))
+  }
+  return result
+}
+
+// Get j-step vector on tonic of `scale` and return it as an array of [String, number] pairs
+// (sorted in descending order by size).
+export function getStepVector(scale: Scale, intervalClass: number): [[String, number]] {
+  if (intervalClass === 0) {
+    return [] // [] represents the unison. If anything else is returned it's a bug.
+  }
+  const scaleAsSteps = [...Array(scale.intervals.length).keys()].map((i) =>
+    scale.rotate(i).getInterval(1)
+  ) // the scale as steps
+  const sortedUniqueSteps = scaleAsSteps.filter(function (item, pos) {
+    // This function asks, "Is it the first `pos` such that `rep(item)` is that value?"
+    // (I had to do this because of the whole Proxy vs. Object weirdness.)
+    return scaleAsSteps.findIndex((step) => step.toString() === item.toString()) == pos
+  })
+  // Sort the scale steps
+  sortedUniqueSteps.sort().reverse() // sort in descending order
```

Comment:
`Array.sort` is a trap. You need to do `foo.sort((a, b) => a.compare(b))` or whatever I had implemented in `scale-workshop-core` if I had implemented anything...